### PR TITLE
Sync recent updates

### PR DESCRIPTION
**Pull request by nshahan**
This bumps the dependency to angular: 5.0.0-alpha+5 which removed the angular transformer. Now the only way to build angular projects is with the build_runner package.

`pub run build_runner build` 
or 
`pub run build_runner serve`

See https://github.com/dart-lang/angular_components_example/tree/deps-bump for an example usage.

There is no longer any need for dependency overrides on analyzer because version  0.31.0+1 has been published.

### CODE REVIEW

**Comment by nshahan** [comment]
Fixes #214

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,8 +1,37 @@
+## 0.9.0-alpha+4
+
+> NOTE: This code is considered production quality, but depends on angular:
+> 5.0.0-alpha+4. The alpha tag represents the evolving nature of the AngularDart
```
Comment (chalin) [inline CHANGELOG.md]: alpha+4 -> alpha+5

File: README.md [hunk 2]
```diff
@@ -40,27 +28,36 @@ and we will work with you.
...
+code generation through package [build]. To build your project with
+[build_runner]:
 
-1. Add two `dev_dependencies` to your packages pubspec.yaml file:
-  ```
+1. Add two dev dependencies to your `pubspec.yaml` file:
+
+  ```yaml
   dev_dependencies:
     build_runner: ^0.7.0
```
Comment (chalin) [inline README.md]: I think that maybe this needs to be ^0.7.8?

File: README.md [hunk 3]
```diff
@@ -40,27 +28,36 @@ and we will work with you.
...
     build_web_compilers: ^0.1.1
```
Comment (chalin) [inline README.md]: This also might need to be updated to ^0.2.1?

File: lib/src/material_tooltip/tooltip_source.dart [hunk 4]
```diff
@@ -24,16 +24,22 @@ const tooltipShowDelay = const Duration(milliseconds: 600);
...
-  'aria-label': 'tooltipLabel',
-  '[attr.style]': "'cursor: pointer'"
-})
+@Directive(
+  selector: '[tooltipSource]',
+  exportAs: 'tooltipSource',
+  host: const {
+    '(mouseover)': 'onMouseOver()',
+    '(mouseleave)': 'onMouseLeave()',
+    '(focus)': r'onFocus()',
```
Comment (chalin) [inline lib/src/material_tooltip/tooltip_source.dart]: Probably don't need the raw string qualifier here.

File: lib/src/material_tree/group/material_tree_group.dart [hunk 5]
```diff
@@ -125,8 +157,49 @@ class MaterialTreeGroupComponent extends MaterialTreeNode implements OnDestroy {
...
+  @override
+  void ngOnDestroy() {
     onDestroy();
   }
+
+  final viewMoreMsg = _viewMoreMsg;
+
+  static String get _viewMoreMsg => Intl.message('View more',
+      desc: 'Label for a link that allows user to see the collapsed options.'
+          '[REL_NOTE: jiahaowang/LAUNCHING-2018-03-31]');
```
Comment (chalin) [inline lib/src/material_tree/group/material_tree_group.dart]: Was the intent really to end the `desc` string with `'[REL_NOTE: jiahaowang/LAUNCHING-2018-03-31]'`?

Comment (nshahan) [reply lib/src/material_tree/group/material_tree_group.dart]: Thanks, I'll follow up with the author of this change to see what was going on here.

File: lib/src/material_tooltip/tooltip_source.dart (hunk 4 above)
Comment (nshahan) [reply lib/src/material_tooltip/tooltip_source.dart]: We are starting to clean up the `host:` properties of the `@Component` and `@Directive` annotations in the library so these will probably go away in the near future.

File: CHANGELOG.md (hunk 1 above)
Comment (nshahan) [reply CHANGELOG.md]: Done.

File: README.md (hunk 2 above)
Comment (nshahan) [reply README.md]: Done.

File: README.md (hunk 3 above)
Comment (nshahan) [reply README.md]: Done.

File: lib/src/material_tree/group/material_tree_group.dart (hunk 5 above)
Comment (nshahan) [reply lib/src/material_tree/group/material_tree_group.dart]: Removed.

**Pull request merged**